packages/quantic/force-app/main/default/lwc/quanticResultCopyToClipboard/__tests__/quanticResultCopyToClipboard.test.js

```diff
@@ -0,0 +1,300 @@
+/* eslint-disable no-import-assign */
+import QuanticResultCopyToClipboard from '../quanticResultCopyToClipboard';
+import {cleanup, buildCreateTestComponent} from 'c/testUtils';
+import * as mockHeadlessLoader from 'c/quanticHeadlessLoader';
+import * as quanticUtils from 'c/quanticUtils';
+
+const defaultTemplate = '${title}\n${clickUri}';
+
+jest.mock('c/quanticHeadlessLoader');
+jest.mock('c/quanticUtils', () => ({
+  ...jest.requireActual('c/quanticUtils'),
+  copyToClipboard: jest.fn(() => Promise.resolve()),
+  buildTemplateTextFromResult: jest.fn(() => defaultTemplate),
+}));
+
+let isInitialized = false;
+const functionMocks = {
+  logCopyToClipboard: jest.fn(() => Promise.resolve()),
+  dispatch: jest.fn(() => {}),
+};
+
+const exampleEngine = {
+  id: 'exampleEngineId',
+  dispatch: functionMocks.dispatch,
+};
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  result: {
+    clickUri: 'https://example.com',
+    title: 'Example Title',
+  },
+  label: 'Copy',
+  successLabel: 'Copied',
+  textTemplate: defaultTemplate,
+};
+
+const copyToClipboardEvent = new CustomEvent('quantic__copytoclipboard', {
+  bubbles: true,
+});
+
+const selectors = {
+  initializationError: 'c-quantic-component-error',
+  quanticResultAction: 'c-quantic-result-action',
+};
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticResultCopyToClipboard,
+  'c-quantic-result-copy-to-clipboard',
+  defaultOptions
+);
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      loadInsightAnalyticsActions: jest.fn(() => {
+        return {
+          logCopyToClipboard: functionMocks.logCopyToClipboard,
+        };
+      }),
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticResultCopyToClipboard && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticResultCopyToClipboard) {
+      element.setInitializationError();
+    }
+  };
+}
+
+async function flushPromises() {
```

Comment:
use utils! :D

Reply:
no this one is an override on purpose, the implementation is different because of the jest timers....